misc-scripts/xref_mapping/XrefParser/RefSeqGPFFParser.pm

```diff
@@ -17,340 +17,330 @@ limitations under the License.
-      else{
-         print "$acc NE $acc_no_ver\n";
+      foreach my $wiki_id (@{ $self->{wiki_ids}->{$gene_id} }) {
+        $self->add_dependent_xref({
+          master_xref_id => $refseq_acc,
+          acc            => $wiki_id,
+          source_id      => $self->source_id_from_name('WikiGene'),
+          species_id     => $self->{species_id},
+          dbi            => $self->{dbi}
+        });
       }
+    }
+  }
 
-      $xref->{LABEL} = $acc . "\." . $ver;
-      $xref->{DESCRIPTION} = $description;
-      $xref->{SOURCE_ID} = $source_id;
-      $xref->{SEQUENCE} = $parsed_seq;
-      $xref->{SEQUENCE_TYPE} = $type;
-      $xref->{SPECIES_ID} = $species_id;
-      $xref->{INFO_TYPE} = "SEQUENCE_MATCH";
+  return $xref;
 
-      # TODO experimental/predicted
+}
 
-      my @EntrezGeneIDline = $entry =~ /db_xref=.GeneID:(\d+)/g;
-#      my @SGDGeneIDline = $entry =~ /db_xref=.SGD:(S\d+)/g;
-      my @protein_id = $entry =~ /\/protein_id=.(\S+_\d+)/g;
-      my @coded_by = $entry =~  /\/coded_by=.(\w+_\d+)/g;
 
-      foreach my $cb (@coded_by){
-	$xref->{PAIR} = $cb;
-      }
-      if (!defined $xref->{PAIR}) {
-        $xref->{PAIR} = $refseq_pair;
-      }
+# returns the source id for a source name, requires $self->{source_ids} to have been populated
+sub source_id_from_name {
+  my ($self, $name) = @_;
 
-      foreach my $pi (@protein_id){
-	$xref->{PROTEIN} = $pi;
-      }
+  my $source_id;
 
-      foreach my $ll (@EntrezGeneIDline) {
-	my %dep;
-        #my $entrez_source = $dependent_sources{EntrezGene} || die( 'No source for EntrezGene!' );
-        #my $wiki_source = $dependent_sources{WikiGene} || die( 'No source for WikiGene!' );
-        if (defined $entrez{$ll}) {
-          $dep{SOURCE_ID} = $entrez_source_id;
-	  $dep{LINKAGE_SOURCE_ID} = $source_id;
-	  $dep{ACCESSION} = $ll;
-          $dep{LABEL} = $entrez{$ll};
-	  push @{$xref->{DEPENDENT_XREFS}}, \%dep;
-
-  	  my %dep2;
-	  $dep2{SOURCE_ID} = $wiki_source_id;
-	  $dep2{LINKAGE_SOURCE_ID} = $source_id;
-	  $dep2{ACCESSION} = $ll;
-          $dep2{LABEL} = $entrez{$ll};
-	  push @{$xref->{DEPENDENT_XREFS}}, \%dep2;
-
-          # Add xrefs for RefSeq mRNA as well where available
-          $refseq_pair =~ s/\.[0-9]*// if $refseq_pair;
-          if (defined $refseq_pair) {
-            if ($refseq_ids->{$refseq_pair}) {
-              foreach my $refseq_id (@{ $refseq_ids->{$refseq_pair} }) {
-                foreach my $entrez_id (@{ $entrez_ids->{$ll} }) {
-                  $add_dependent_xref_sth->execute($refseq_id, $entrez_id);
-                }
-                foreach my $wiki_id (@{ $wiki_ids->{$ll} }) {
-                  $add_dependent_xref_sth->execute($refseq_id, $wiki_id);
-                }
-              }
-            }
-          }
-        }
-      }
+  if ( exists $self->{source_ids}->{$name} ) {
+    $source_id = $self->{source_ids}->{$name};
+  } elsif ( $self->{verbose} ) {
+    warn "WARNING: can't get source ID for name '$name'\n";
```

Comment:
Is this really a non-fatal thing? Feels like it should be an error.

Reply:
@tgrego ping. Moreover, if it really **is** non-fatal, shouldn't we use carp() instead of warn()?

Reply:
ok, I feel it's safer to actually have it a critical error, so now confesses